### PR TITLE
BFN platform API 2.0 support

### PR DESCRIPTION
**Pull request by vboykox**
**- Why I did it**
Added barefoot platform api 2.0 support

**- Description for the changelog**
Added barefoot platform api 2.0 support

### CODE REVIEW

**Comment by akokhan** [comment]
By adding this, I believe you should remove platform 1.0 plugins.

**Comment by vboykox** [comment]
> By adding this, I believe you should remove platform 1.0 plugins.

decode-eeprom, sfputil, psuutil still use API 1.0

**Comment by vboykox** [comment]
Blocked until https://github.com/Azure/sonic-platform-common/pull/96 is merged and sonic-buildimage updated correspondingly.

**Comment by lguohan** [comment]
https://github.com/Azure/sonic-platform-common/pull/96 is merged. can you fix the lgtm alerts?

**Comment by vboykox** [comment]
LGTM

**Comment by vboykox** [comment]
lgtm

**Comment by vboykox** [comment]
retest this please

**Comment by vboykox** [comment]
> * 2 for Wrong number of arguments in a class instantiation

It looks like the check was done against corresponding constructor from other vendors' code, please ignore.

**Comment by vboykox** [comment]
@lguohan the PR has been updated

**Comment by vboykox** [comment]
@lguohan

**Comment by lguohan** [comment]
can you help to explain the alerts "2 for Wrong number of arguments in a class instantiation"?

**Comment by vboykox** [comment]
> > * 2 for Wrong number of arguments in a class instantiation
> 
> It looks like the check was done against corresponding constructors from other vendors' code, please ignore.

@lguohan  False positive; the warning was also ignored for other verndors' PRs

**Comment by vboykox** [comment]
@lguohan could you please merge

**Comment by vboykox** [comment]
@lguohan

**Comment by vboykox** [comment]
https://github.com/Azure/sonic-buildimage/pull/4552#issuecomment-628897195

@lguohan please checkout the comment from merged PR.

**Comment by vboykox** [comment]
retest mellanox please

**Comment by vboykox** [comment]
@lguohan

**Comment by vboykox** [comment]
retest mellanox please

**Comment by vboykox** [comment]
retest baseimage please

**Comment by akokhan** [comment]
@lguohan, please approve and merge

**Comment by vboykox** [comment]
@lguohan could you please give a reply?

**Comment by vboykox** [comment]
@lguohan please merge

**Comment by vboykox** [comment]
@lguohan @jleveque

**Comment by jleveque** [comment]
@vboykox; Please fix conflicts.

**Comment by vboykox** [comment]
@jleveque done

**Review by jleveque — CHANGES_REQUESTED** [review]
There are still LGTM alerts. Please fix.

**Comment by vboykox** [comment]
> > > * 2 for Wrong number of arguments in a class instantiation
> > 
> > 
> > It looks like the check was done against corresponding constructors from other vendors' code, please ignore.
> 
> @lguohan False positive; the warning was also ignored for other verndors' PRs

@jleveque

**Comment by vboykox** [comment]
> [#4552 (comment)](https://github.com/Azure/sonic-buildimage/pull/4552#issuecomment-628897195)
> 
> @lguohan please checkout the comment from merged PR.

@jleveque

**Comment by jleveque** [comment]
Thanks for pointing that out. Looks like the LGTM tool needs some imporvements :)

File: platform/barefoot/sonic-platform-modules-bfn/debian/rules [hunk 1]
```diff
@@ -3,20 +3,32 @@
...
 CONFIGS_SRC := $(shell pwd)/configs
+BUILD_DIR := $(shell pwd)/build
+WHEEL_BUILD_DIR := $(BUILD_DIR)/wheel
 
 %:
 	dh $@
 
+override_dh_auto_build:
+	set -e
+	python2.7 setup.py bdist_wheel -d $(WHEEL_BUILD_DIR)
```
Comment (jleveque) [inline platform/barefoot/sonic-platform-modules-bfn/debian/rules:14]: We are currently in the process of transitioning all Python code from Python 2 to Python 3. We are now asking all vendors to build/install/copy both Python 2 and Python 3 versions of their sonic_platform package in preparation for the transition. Can you please update the PR to do this?

**Review by jleveque — CHANGES_REQUESTED** [review]

File: platform/barefoot/sonic-platform-modules-bfn/debian/rules (hunk 1 above)
Comment (vboykox) [reply platform/barefoot/sonic-platform-modules-bfn/debian/rules:14]: Done

**Comment by jleveque** [comment]
@vboykox: Thank you very much for helping us prepare for the transition to Python 3!

**Review by jleveque — APPROVED** [review]

**Pull request merged**